Add Send email feature

quiteqiang opened the pull request:
Add send email feature to single and multiple people from one group
Fixed the add member button function

further goal: need to update the path to get ManagerEmail
Note: update the email sender in the email-api file and username and password in the private file are required to send an email successfully

Commit:
Add Send email feature

Add send email feaure to singlr and multiple people from one group
Fixed the add member button funtion

furtuher goal: need to update the path to get ManagerEmail

PROgram52bc commented on api/src/emails/api.py:
```diff
@@ -21,11 +21,14 @@ def send_email():
-                                                  bcc=valid_email_request['bcc']);
+
+    msg = Message(valid_email_request['subject'], recipients=valid_email_request['recipients'],
+                                                   sender="qiangwang121212@gmail.com")
```
We might want to encode the sender address from the client here.

PROgram52bc commented on api/src/emails/api.py:
```diff
@@ -21,11 +21,14 @@ def send_email():
+
+    msg = Message(valid_email_request['subject'], recipients=valid_email_request['recipients'],
+                                                   sender="qiangwang121212@gmail.com")
+#                                                     valid_email_request['managerEmail']),
+#                                                   valid_email_request['managerName'],
+#                                                   reply_to=valid_email_request['managerEmail'])
```
Remove these? Instead of comment

PROgram52bc commented on api/src/emails/api.py:
```diff
@@ -21,11 +21,14 @@ def send_email():
+#                                                     valid_email_request['managerEmail']),
+#                                                   valid_email_request['managerName'],
+#                                                   reply_to=valid_email_request['managerEmail'])
+#                                                   cc=valid_email_request['cc'],
+#                                                   bcc=valid_email_request['bcc'])
```
Maybe ask Dr. Nurkkala later, if we need those.

PROgram52bc commented on api/src/emails/test_emails.py:
```diff
@@ -23,8 +23,13 @@ def __call__(self):
-        'recipients': ['tim_ours@taylor.edu']
+        'recipients': ['qiang_wang@taylor.edu'],
+        'body': 'This is a test email',
+#         'body': rl_fake().sentences(nb=1)[0],
+#         'managerName': 'qwang',
+#         'managerEmail': 'qiangwang121212@gmail.com',
```
Remove those.

PROgram52bc commented on api/src/emails/test_emails.py:
```diff
@@ -37,8 +42,11 @@ def test_send_email(auth_client):
     # GIVEN nothing
 
     # WHEN we try to send an email
+#     print(url_for('emails.send_email'))
+#     print(email_object_factory())
```
remove these....

PROgram52bc commented on api/src/groups/api.py:
```diff
@@ -427,7 +427,7 @@ def create_member(group_id):
     except ValidationError as err:
         return logged_response(err.messages, 422)
-
+    print(valid_member)
```
remove...

PROgram52bc commented on ui/src/components/groups/members/GroupMembers.vue:
```diff
@@ -251,6 +251,7 @@
           <div>
             <h3 class="headline mb-0">
               {{ $t("groups.members.email.compose") }}
+              <!--    "Compose Emial" title          -->
```
spelling

PROgram52bc commented on ui/src/components/groups/members/GroupMembers.vue:
```diff
@@ -735,6 +735,7 @@ export default {
         this.people = this.members.map(e => e.person);
         this.parseMembers();
         this.tableLoading = false;
+        console.log(resp);
```
remove?...

PROgram52bc reviewed (changes requested):
I guess just two things before this merge:
1. remove unnecessary comments
1. use the "sender" field provided by the client instead of a hard-coded email address.

Commit:
Add Message functuion for managers

Add send message funcion for managers.
Manger send group email fixed.

Additional UI changes to manager page

PROgram52bc commented on ui/i18n/cc-i18n.json:
```diff
@@ -1217,6 +1218,7 @@
       "female": "Mujer",
       "gender": "G\u00e9nero",
       "male": "Hombre",
+      "manager-type":"Tipo de gerente\n",
```
remove \n?

PROgram52bc commented on ui/src/components/groups/GroupManagers.vue:
```diff
@@ -76,7 +76,7 @@
       :rows-per-page-items="rowsPerPageItem"
       :headers="headers"
-      :items="visibleMembers"
+      :items="managers"
```
add `visibleManagers` later

PROgram52bc commented on api/src/emails/test_emails.py:
```diff
@@ -38,7 +40,8 @@ def test_send_email(auth_client):
     # WHEN we try to send an email
     resp = auth_client.post(url_for('emails.send_email'),
-                            json=email_object_factory())
+                            json=email_object_factory(),
+                            )
```
indent

PROgram52bc commented on api/src/emails/api.py:
```diff
@@ -21,11 +21,11 @@ def send_email():
-                                                  cc=valid_email_request['cc'],
-                                                  bcc=valid_email_request['bcc']);
+
+    msg = Message(valid_email_request['subject'], recipients=valid_email_request['recipients'],
```
indent?

PROgram52bc commented on ui/src/components/groups/GroupManagers.vue:
```diff
@@ -530,14 +524,18 @@
       },
 
       getEmailRecipients() {
+        console.log(this.selected);
```
remove

PROgram52bc commented on ui/src/components/groups/GroupManagers.vue:
```diff
@@ -530,14 +524,18 @@
         return this.selected
           .map(e => e.person.email)
           .filter(function(e) {
+            console.log(e);
```
remove

PROgram52bc commented on ui/src/components/groups/GroupManagers.vue:
```diff
@@ -530,14 +524,18 @@
 
       sendEmail() {
+        this.email['managerEmail'] = this.currentAccount.email;
+        console.log("----",this.email);
```
remove

PROgram52bc commented on ui/src/components/groups/GroupManagers.vue:
```diff
@@ -741,6 +739,7 @@
 
     mounted: function() {
       this.getMembers();
+      console.log(this.currentAccount);
```
remove

PROgram52bc commented on ui/src/pages/Login.vue:
```diff
@@ -99,12 +99,14 @@ export default {
           return;
         } else {
           const decodedJwt = jwtDecode(resp.data.jwt);
+          console.log(resp.data)
```
remove

PROgram52bc commented on ui/src/components/groups/GroupManagers.vue:
```diff
@@ -731,8 +728,9 @@
-          this.people = this.members.map(e => e.person);
+          this.managers = resp.data;
+          this.people = this.managers.map(e => e.person);
+          // console.log(this.people);
```
remove

Commit:
Remove unreleated comment

Clean up the code base

PROgram52bc commented on ui/src/components/groups/GroupManagers.vue:
```diff
@@ -504,11 +498,12 @@
         this.addParticipantDialog.loading = true;
         let promises = [];
-
+        console.log(this.addParticipantDialog.newParticipants);
```
remove this

PROgram52bc commented on ui/src/components/groups/GroupManagers.vue:
```diff
@@ -504,11 +498,12 @@
+          const idx = this.managers.findIndex(
             gr_pe => gr_pe.person.person_id === person.id
           );
+          console.log(person.id);
```
remove this

PROgram52bc commented on ui/src/components/groups/GroupManagers.vue:
```diff
@@ -565,15 +559,14 @@
+        };
+        return this.$http.post(`/api/v1/groups/groups/${ groupId }/managers`, {
+          personId: id,
+          managerTypeId: 1
```
`managerTypeId` shouldn't be hardcoded

PROgram52bc commented on ui/src/components/groups/members/GroupMembers.vue:
```diff
@@ -504,7 +504,7 @@ export default {
       this.addParticipantDialog.loading = true;
       let promises = [];
-
+      console.log(this.addParticipantDialog.newParticipants);
```
remove this

quiteqiang committed:
unreleated file

Remove unrelated file

Commit:
Remove redundant comment

Remove not related code

Commit:
fixed merge error

Commit:
fix send group email issue

fixed

PROgram52bc committed:
Delete GroupEmail.vue

PROgram52bc reviewed (changes requested):
remove console statement, and it should be good

Commit:
Fix extra informaiton

Fixed

Commit:
Merge branch 'feature/542-group-new-model-ui-send-email' of https://github.com/corpus-christi/corpus-christi into feature/542-group-new-model-ui-send-email
